### PR TITLE
fix: trips changes in dispatch module

### PR DESCRIPTION
**Pull request by FH-sandeepkaur1**
## Change Description

Changes in add/edit of trips module

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist: (Check only applicable)

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by amritfleethawks — APPROVED** [review]

File: src/app/pages/dispatch/permanent-routing/add-route/add-route.component.ts [hunk 1]
```diff
@@ -529,6 +529,7 @@ export class AddRouteComponent implements OnInit {
         this.response = res;
         this.hasSuccess = true;
         this.spinner.hide();
+        this.router.navigateByUrl('/dispatch/routes/list');
         this.toastr.success('Route updated successfully');
```
Comment (kunalfleethawks) [inline src/app/pages/dispatch/permanent-routing/add-route/add-route.component.ts]: please add full stop after sentence.

File: src/app/pages/dispatch/permanent-routing/route-list/route-list.component.ts [hunk 2]
```diff
@@ -67,6 +67,7 @@ export class RouteListComponent implements AfterViewInit, OnDestroy, OnInit {
         this.rerender();
         this.spinner.hide();
         this.hasSuccess = true;
+        this.toastr.success('Route deleted successfully');
```
Comment (kunalfleethawks) [inline src/app/pages/dispatch/permanent-routing/route-list/route-list.component.ts]: please add full stop after sentence.

File: src/app/pages/dispatch/trips/add-trip/add-trip.component.ts [hunk 3]
```diff
@@ -1014,35 +964,429 @@ export class AddTripComponent implements OnInit {
...
+                        error: () => {
+                        },
+                        next: () => {
+                        },
+                    });
+            },
+            next: (res) => {
+                this.spinner.hide();
+                this.response = res;
+                this.toastr.success('Trip updated successfully');
```
Comment (kunalfleethawks) [inline src/app/pages/dispatch/trips/add-trip/add-trip.component.ts]: please add full stop after sentence.

File: src/app/pages/dispatch/trips/trip-list/trip-list.component.ts [hunk 4]
```diff
@@ -268,21 +218,19 @@ export class TripListComponent implements AfterViewInit, OnDestroy, OnInit {
...
+    if (confirm('Are you sure you want to delete?') === true) {
+      this.spinner.show();
+      this.apiService.getData('trips/delete/' + tripID + '/1').subscribe({
+        complete: () => {},
+        error: () => { },
+        next: (result: any) => {
+          this.spinner.hide();
+          this.hasSuccess = true;
+          this.rerender();
+          this.toastr.success('Trip deleted successfully');
```
Comment (kunalfleethawks) [inline src/app/pages/dispatch/trips/trip-list/trip-list.component.ts:230]: please add full stop after sentence.

File: src/app/pages/dispatch/trips/trip-list/trip-list.component.ts [hunk 5]
```diff
@@ -339,8 +286,6 @@ export class TripListComponent implements AfterViewInit, OnDestroy, OnInit {
           });
       },
       next: (res) => {
-        // this.fetchTrips();
-
         this.spinner.hide();
         $("#tripStatusModal").modal('hide');
         this.toastr.success('Trip status updated successfully');
```
Comment (kunalfleethawks) [inline src/app/pages/dispatch/trips/trip-list/trip-list.component.ts]: please add full stop after sentence.

**Review by kunalfleethawks — APPROVED** [review]

**Pull request merged**